### PR TITLE
Resolve nested subschemas from id refs 

### LINKED ISSUE #69: allOf and referenced schemas always returning true for the referenced type
allOf with refs to remote schemas (or resolved local schemas) do not validate referenced types properly
i.e 

```
schemaWithRefs  = {
   "$id"=>"NAME_HERE",
   "$schema"=>"http://json-schema.org/draft-04/schema#",
   "type"=>"object",
   "description"=>"A Host entity has been created or changed",
   "required"=>["host"],
   "additionalProperties"=>false,
   "properties"=> {
      "host"=> {
        "description"=>"Host", 
       "allOf"=>[{"$ref"=>"https://my-url-here.com/EDM.json#Host"}], 
       "required"=>["attribute1"]
      }
   } 
}
```

And the referenced schema fragment being 

```
{
 "Host" => {
   "$id" => "#Host",
   "properties" => { "attribute1" => { "type" => "string"  } }
 }
}

```

```
schema = JSONSchemer.schema(schemaWithRefs, ref_resolver: 'net/http')
schema.valid?({
  'host' => {
   "attribute1" => nil
  }
})
// true
```

### PR DESCRIPTION
**Pull request by Jbern16**
https://json-schema.org/understanding-json-schema/structuring.html#using-id-with-ref

This PR allows id refs to work in nested json structures, as outlined in the spec by building relative json pointers for every ID found in the document. @eraserhd

Fixes #69

### CODE REVIEW

**Comment by Jbern16** [comment]
@davishmcclurg any thoughts? Thanks in advance!

File: lib/json_schemer/schema/base.rb [hunk 1]
```diff
@@ -615,16 +615,17 @@ def resolve_ids(schema, ids = {}, parent_uri = nil, pointer = '')
...
-              schema: schema,
-              pointer: pointer
-            }
-          end
-          if definitions = schema['definitions']
-            definitions.each { |key, subschema| resolve_ids(subschema, ids, uri, "#{pointer}/definitions/#{key}") }
+          uri = join_uri(parent_uri, schema[id_keyword])
+          schema.each do |key, value| 
+            if key == id_keyword
+              unless uri == parent_uri
```
[the remark this answers is not in the document]
Comment (Jbern16) [reply lib/json_schemer/schema/base.rb]: Updated!

**Comment by Jbern16** [comment]
Awesome. Let me know if there is anything more I can do get this moved along

File: lib/json_schemer/schema/base.rb [hunk 2]
```diff
@@ -615,16 +615,15 @@ def resolve_ids(schema, ids = {}, parent_uri = nil, pointer = '')
...
-              schema: schema,
-              pointer: pointer
-            }
-          end
-          if definitions = schema['definitions']
-            definitions.each { |key, subschema| resolve_ids(subschema, ids, uri, "#{pointer}/definitions/#{key}") }
+          uri = join_uri(parent_uri, schema[id_keyword])
+          schema.each do |key, value| 
+            if key == id_keyword &&  uri != parent_uri
+                ids[uri.to_s] = {
```
Comment (davishmcclurg) [inline lib/json_schemer/schema/base.rb:621]: Really minor, but this has some extra indentation now

Comment (Jbern16) [reply lib/json_schemer/schema/base.rb]: updated

**Comment by davishmcclurg** [comment]
Thanks @Jbern16!

**Comment by davishmcclurg** [comment]
Released in 0.2.13

**Pull request merged**